File: src/AxisCRM.Api/Domain/Models/Usuario.cs
using AxisCRM.Api.Domain.Enums;

namespace AxisCRM.Api.Domain.Models
{
    public class Usuario : EntidadeBase
    {
        public string Email { get; set; }
        public string Senha { get; set; }
        public bool Excluido { get; set; } = false;
        public DateTime? DataExclusao { get; set; }
        public PerfilUsuario Perfil { get; set; } = PerfilUsuario.Padrao;
    }
}

File: App/src/models/Usuario.js
export default class Usuario {
  constructor({
    id = 0,
    email = '',
    senha = '',
    perfil = null,
    dataCadastro = null,
    excluido = false,
    dataExclusao = null
  } = {}) {
    this.id = id;
    this.email = email;
    this.senha = senha;
    this.perfil = perfil;
    this.dataCadastro = dataCadastro
      ? new Date(dataCadastro)
      : null;
    this.excluido = excluido;
    this.dataExclusao = dataExclusao
      ? new Date(dataExclusao)
      : null;
  }

  modeloValidoLogin() {
    return Boolean(this.email && this.senha);
  }
}
